src/core/vector/qgsvectorlayereditbuffergroup.cpp

```diff
@@ -382,93 +389,40 @@ bool QgsVectorLayerEditBufferGroup::isEditing() const
 
 QList<QgsVectorLayer *> QgsVectorLayerEditBufferGroup::orderLayersParentsToChildren( QSet<QgsVectorLayer *> layers )
 {
-  QSet<QgsVectorLayer *> referencingLayers;
-  QSet<QgsVectorLayer *> referencedLayers;
-
-  {
-    const QList<QgsRelation> relations = QgsProject::instance()->relationManager()->relations().values();
-    for ( const QgsRelation &relation : relations )
-    {
-      referencingLayers.insert( relation.referencingLayer() );
-      referencedLayers.insert( relation.referencedLayer() );
-    }
-  }
-
   QList<QgsVectorLayer *> orderedLayers;
+  QSet<QgsVectorLayer *> unorderedLayers = layers;
 
-  // Layers that are only parents
+  bool layerOrdered = true;
+  while ( ! unorderedLayers.isEmpty() || layerOrdered )
```

Comment:
Should this be `&&`? If this is empty, there's no reason to continue even if `layerOrdered` is true? And if `layerOrdered` is false we risk an endless loop?

Reply:
Yes thanks it should clearly be an `&&` 🙈